packages/grid_client/tests/modules/applications/gitea.test.ts

```diff
@@ -0,0 +1,212 @@
+import axios from "axios";
+import { setTimeout } from "timers/promises";
+
+import { FilterOptions, GatewayNameModel, generateString, GridClient, MachinesModel, randomChoice } from "../../../src";
+import { config, getClient } from "../../client_loader";
+import { GBToBytes, generateInt, getOnlineNode, log, RemoteRun, splitIP } from "../../utils";
+
+jest.setTimeout(1250000);
+
+let gridClient: GridClient;
+let deploymentName: string;
+
+beforeAll(async () => {
+  gridClient = await getClient();
+  deploymentName = "gt" + generateString(10);
+  gridClient.clientOptions.projectName = `gitea/${deploymentName}`;
+  gridClient._connect();
+  return gridClient;
+});
+
+// Private IP Regex
+const ipRegex = /(^127\.)|(^10\.)|(^172\.1[6-9]\.)|(^172\.2[0-9]\.)|(^172\.3[0-1]\.)|(^192\.168\.)/;
+
+test("TC2954 - Applications: Deploy Gitea", async () => {
+  /**********************************************
+     Test Suite: Grid3_Client_TS (Automated)
+     Test Cases: TC2954 - Applications: Deploy Gitea
+     Scenario:
+        - Generate Test Data/Gitea Config/Gateway Config.
+        - Select a Node To Deploy the Gitea on.
+        - Select a Gateway Node To Deploy the gateway on.
+        - Deploy the Gitea solution.
+        - Assert that the generated data matches
+          the deployment details.
+        - Pass the IP of the Created Gitea to the Gateway
+          Config.
+        - Deploy the Gateway.
+        - Assert that the generated data matches
+          the deployment details.
+        - Assert that the Gateway points at the IP
+          of the created Gitea.
+        - Assert that the returned domain is working
+          and returns correct data.
+    **********************************************/
+  const name = "gw" + generateString(10).toLowerCase();
+  const tlsPassthrough = false;
+  const cpu = 1;
+  const memory = 2;
+  const rootfsSize = 2;
+  const diskSize = 15;
+  const networkName = generateString(15);
+  const vmName = generateString(15);
+  const diskName = generateString(15);
+  const mountPoint = "/var/lib/docker";
+  const publicIp = false;
+  const ipRangeClassA = "10." + generateInt(1, 255) + ".0.0/16";
+  const ipRangeClassB = "172." + generateInt(16, 31) + ".0.0/16";
+  const ipRangeClassC = "192.168.0.0/16";
+  const ipRange = randomChoice([ipRangeClassA, ipRangeClassB, ipRangeClassC]);
+  const metadata = "{'deploymentType': 'gitea'}";
+  const description = "test deploying Gitea via ts grid3 client";
+
+  // Gateway Node Selection
+  const gatewayNodes = await gridClient.capacity.filterNodes({
+    gateway: true,
```

Comment:
add features in the filters to avoid retrieving nodes that don't align with required features